### ISSUE TITLE
How to send selection nodes name to app.tsx

### ISSUE THREAD

**Issue by PepperKUN** [CoconutGoodie/figma-plugin-react-vite#6]
Hello, I am now in a situation of how to return the node name string selected by figma to the App.tsx page. So far, it seems that I can only handle this string information in the class under messages. 
What if I want to return these strings to App.tsx and use setState to refresh the display of the page. 
How should it be done according to the current structure?

**Comment by PepperKUN** [comment]
```typescript
import * as Networker from "monorepo-networker";
import { NetworkSide } from "../sides";
import { keyframe } from "../../../type/types";

interface Payload {
  keyframes: {
    [key:string]: keyframe[]
  };
  allowPlay: boolean;
}

export class SelectionMessage extends Networker.MessageType<Payload> {

  receivingSide(): Networker.Side {
    return NetworkSide.UI;
  }

  handle(payload: Payload) {
    //can only resolve the message here
    console.group('selection')
    console.log(payload.allowPlay)
    console.table(payload.keyframes);
    console.groupEnd()
  }
}
```

**Comment by iGoodie** [comment]
Hey there! 😄 

In theory, you can use a global store library (like Redux or Zustand) to store your data globally and initiate state updates outside a React component. Theoretically it'd look like so:

- plugin/plugin.ts
```ts
    figma.on("selectionchange", () => {
      const { selection } = figma.currentPage;

      NetworkMessages.SELECTION_CHANGED.send({
        selection: selection.map((selection) => ({
          id: selection.id,
          boundingBox: selection.absoluteBoundingBox,
        })),
      });
    });
```

- common/network/messages/SelectionMessage.ts
```ts
import { store } from "/path/to/createdStore";

interface Payload {
  selection: {
    id: string;
    boundingBox: Rect | null;
  }[];
}

export class SelectionMessage extends Networker.MessageType<Payload> {
  receivingSide(): Networker.Side<any> {
    return NetworkSide.UI;
  }

  handle(payload: Payload): void {
    store.dispatch({
      type: "SELECT",
      payload,
    });
  }
}
```

**Comment by iGoodie** [comment]
I might also want to support arbitrary subscriptions to the messages too.
I'll include this under https://github.com/CoconutGoodie/monorepo-networker as a feature request 😄

**Comment by PepperKUN** [comment]
Thank you, using a global store library is better than my current method. I'll try this way to assign data from figma to plugin page. I also looking forward to the new feature, it seems to make the code concise and elegant. 😍

**Comment by rocky-at-planningo** [comment]
> Hey there! 😄
> 
> In theory, you can use a global store library (like Redux or Zustand) to store your data globally and initiate state updates outside a React component. Theoretically it'd look like so:
> 
> * plugin/plugin.ts
> 
> ```ts
>     figma.on("selectionchange", () => {
>       const { selection } = figma.currentPage;
> 
>       NetworkMessages.SELECTION_CHANGED.send({
>         selection: selection.map((selection) => ({
>           id: selection.id,
>           boundingBox: selection.absoluteBoundingBox,
>         })),
>       });
>     });
> ```
> 
> * common/network/messages/SelectionMessage.ts
> 
> ```ts
> import { store } from "/path/to/createdStore";
> 
> interface Payload {
>   selection: {
>     id: string;
>     boundingBox: Rect | null;
>   }[];
> }
> 
> export class SelectionMessage extends Networker.MessageType<Payload> {
>   receivingSide(): Networker.Side<any> {
>     return NetworkSide.UI;
>   }
> 
>   handle(payload: Payload): void {
>     store.dispatch({
>       type: "SELECT",
>       payload,
>     });
>   }
> }
> ```

Great work, much appreciated!

**Comment by iGoodie** [comment]
The new API I crafted looks like so, I think it should finally solve this issue :P
I'll be finalizing and merging it in a few days 🎉 

https://github.com/CoconutGoodie/monorepo-networker/blob/ac6e86990985db816c0588e50d1484cd5c8e30f0/examples/pseudo-sides-example/packages/client/main.ts#L19-L24

**Comment by iGoodie** [comment]
It is finally implemented and released with #12 🎉 😄 
Now you're able to subscribe for responses anywhere you want :P

Code from the updated template source: 
https://github.com/CoconutGoodie/figma-plugin-react-vite/blob/8da992e215e4524ee6bb93cebf4501329c98d2bc/src/ui/app.tsx#L17-L21
